pkg/controller/integration/kits.go

```diff
@@ -82,6 +82,18 @@ func lookupKitsForIntegration(ctx context.Context, c ctrl.Reader, integration *v
 	return kits, nil
 }
 
+// sameOrMatch returns whether the v1.IntegrationKit ios the one used by the v1.Integration or if it meets the
+// requirements of the v1.Integration.
+func sameOrMatch(kit *v1.IntegrationKit, integration *v1.Integration) (bool, error) {
```

Comment:
I think this func is a necessary one to have, but also wonder if `integrationMatches()` should be aware of the traits in annotations as well. Currently `integrationMatches()` only compares `it.spec.traits` with `kit.spec.traits` but shouldn't it really compare the combination of annotations and spec.traits?

Reply:
agree, see the other comments